app/utils/report/test.py

```diff
@@ -114,6 +111,11 @@ def _add_test_group_data(group, db, spec, hierarchy=[]):
                 regr_count += 1
         test_cases.append(test_case)
 
+    try:
+        test_cases.sort(key=lambda tc: tc[models.INDEX_KEY])
+    except KeyError:
+        utils.LOG.warning('Index not found for test cases, leaving unsorted.')
```

Comment:
Is this only to be able to deal with older test cases which don't have an `index` field, or is it expected that new test cases also may not have it?

Reply:
New test cases should always have the `index` field. This is just for the older TCs which may lack one.

Reply:
OK, then I think we can simplify the code and remove the `try`-`except` clauses as there shouldn't be email reports generated now with old test data.